connect-inject/health_check_resource_ent_test.go

```diff
@@ -0,0 +1,242 @@
+// +build enterprise
+
+package connectinject
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testNamespace               = "testnamespace"
+	testNamespacedHealthCheckID = "testnamespace_test-pod-test-service_kubernetes-health-check-ttl"
+)
+
+var ignoredFieldsEnterprise = []string{"Node", "Definition", "ServiceID", "ServiceName"}
+
+var testPodWithNamespace = corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: testNamespace,
+		Name:      testPodName,
+	},
+	Spec: corev1.PodSpec{},
+}
+
+func TestReconcilePodWithNamespace(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		Name                 string
+		PreCreateHealthCheck bool
+		InitialState         string
+		Pod                  *corev1.Pod
+		Expected             *api.AgentCheck
+	}{
+		{
+			Name:                 "reconcilePod will create check and set passed",
+			PreCreateHealthCheck: false,
+			InitialState:         "", // only used when precreating a health check
+			Pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: testNamespace,
```

Comment:
should we make the kube namespace different in these tests? I think because they are the same, we don't know if the code is looking at the namespace or the annotation.

Reply:
Great point! Thanks!!
I've added an additional test which covers this as a scenario.